compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -2233,21 +2234,59 @@ public void visit(BLangTableConstructorExpr tableConstructorExpr, AnalyzerData d
     @Override
     public void visit(BLangRecordLiteral recordLiteral, AnalyzerData data) {
         List<RecordLiteralNode.RecordField> fields = recordLiteral.fields;
+        BType recLiteralType = recordLiteral.getBType();
 
         for (RecordLiteralNode.RecordField field : fields) {
             if (field.isKeyValueField()) {
                 analyzeExpr(((BLangRecordKeyValueField) field).valueExpr, data);
+
+                BVarSymbol fieldSymbol = ((BLangRecordKeyValueField) field).key.fieldSymbol;
+                if (fieldSymbol != null) {
+                    reportIfDeprecatedUsage(fieldSymbol, fieldSymbol.toString(),
+                            recordLiteral, ((BLangRecordKeyValueField) field).pos);
+                }
             } else if (field.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
-                analyzeExpr((BLangRecordLiteral.BLangRecordVarNameField) field, data);
+                BLangRecordLiteral.BLangRecordVarNameField recField
+                        = (BLangRecordLiteral.BLangRecordVarNameField) field;
+                analyzeExpr(recField, data);
+
+                if (recLiteralType.tsymbol != null && recLiteralType.tsymbol.type != null
+                        && recLiteralType.tsymbol.type.getKind() == TypeKind.RECORD) {
```

Comment:
We are not querying in key-value fields as in here `recordType.getFields().get(recField.symbol.getName().getValue())`. Yeah, this is repeating unnecessarily, correcting it.

Reply:
Shall we extract fields out to a variable outside anyway to avoid accessing per variable name field? Can be null, but we can just do one null check.

Alternatively, we can collect all the field names (along with pos) and do the deprecation log outside.

Reply:
Fixed via https://github.com/ballerina-platform/ballerina-lang/pull/40545/commits/94071a8a0a22c964a884c40be575bd9193be8a7b